clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/GoogleExecutorTraits.groovy

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.google
+
+import com.google.api.client.googleapis.batch.BatchRequest
+import com.google.api.client.googleapis.services.AbstractGoogleClientRequest;
+import com.netflix.spectator.api.Clock
+import com.netflix.spectator.api.Id
+import com.netflix.spectator.api.Registry
+
+import java.io.IOException
+import java.util.concurrent.TimeUnit;
+
+
+trait GoogleExecutorTraits {
+  final String TAG_BATCH_CONTEXT = "context"
+  final String TAG_REGION = "region"
+  final String TAG_SCOPE = "scope"
+  final String TAG_ZONE = "zone"
+  final String SCOPE_BATCH = "batch"
+  final String SCOPE_GLOBAL = "global"
+  final String SCOPE_REGIONAL = "regional"
+  final String SCOPE_ZONAL = "zonal"
+
+  abstract Registry getRegistry()
+
+  public void timeExecuteBatch(BatchRequest batch, String batchContext, String... tags) throws IOException {
+     Registry registry = getRegistry()
+     Clock clock = registry.clock()
+     Id id = registry.createId("google.batch.execute", tags).withTags(TAG_BATCH_CONTEXT, batchContext, TAG_SCOPE, SCOPE_BATCH)
+     Id sizeId = registry.createId("google.batchSize", tags).withTag(TAG_BATCH_CONTEXT, batchContext)
+
+     long startTime = clock.monotonicTime()
+
+     try {
+       batch.execute()
+       long nanos = clock.monotonicTime() - startTime
+       registry.timer(id.withTag("success", "true")).record(nanos, TimeUnit.NANOSECONDS)
+       registry.counter(sizeId.withTag("success", "true")).increment(batch.size())
+     } catch (IOException ioex) {
+       registry.timer(id.withTag("success", "false")).record(clock.monotonicTime() - startTime, TimeUnit.NANOSECONDS)
+       registry.counter(sizeId.withTag("success", "false")).increment(batch.size())
+       throw ioex
+     } catch (Exception ex) {
+       registry.timer(id.withTag("success", "false")).record(clock.monotonicTime() - startTime, TimeUnit.NANOSECONDS)
+       registry.counter(sizeId.withTag("success", "false")).increment(batch.size())
+       throw new IllegalStateException(ex);
```

Comment:
If you've seen a similar pattern elsewhere in the neighborhood, I can live with it. Maybe `RuntimeException` would work too. I think you just want any unchecked exception here, and `RuntimeException` is the most general of those.